TRLevelControl/Model/Common/FloorData/FDControl.cs

```diff
@@ -172,6 +172,26 @@ public void RemoveEntityTriggers(int entityIndex)
         }
     }
 
+    public List<short> GetTriggerRooms<R>(int entityIndex, List<R> rooms)
+        where R : TRRoom
+    {
+        List<FDTriggerEntry> triggers = GetEntityTriggers(entityIndex);
+        List<short> triggerRooms = new();
+        for (short i = 0; i < rooms.Count; i++)
+        {
+            foreach (TRRoomSector sector in rooms[i].Sectors.Where(s => s.FDIndex != 0))
+            {
+                if (triggers.Any(_entries[sector.FDIndex].Contains))
+                {
+                    triggerRooms.Add(i);
+                    break;
+                }
+            }
+        }
+
+        return triggerRooms;
```

Comment:
This could be a fairly simple looking Linq one-liner if `Room` knew what number room it was - but perhaps something for the future.

Reply:
Yes, that would be good to implement. I don't think we shuffle rooms around anywhere so it's probably not something that would need management, but can have a think about it for the future. I've tidied it up for now anyway with some Linq so it's less loopy, and there is a test to cover it as well.